src/main/java/io/cdap/plugin/gcp/bigquery/util/BigQueryUtil.java

```diff
@@ -180,11 +184,35 @@ public static void validateFieldSchemaMatches(Field bqField, Schema.Field field,
       return;
     }
 
-    if (TYPE_MAP.get(type) == null) {
+    if (TYPE_MAP.get(type) == null && !SUPPORTED_COMPLEX_TYPES.contains(type)) {
       throw new IllegalArgumentException(String.format("Field '%s' is of unsupported type '%s'",
                                                        field.getName(), type));
     }
 
+    if (type == Schema.Type.ARRAY) {
+      String name = field.getName();
+      if (field.getSchema().isNullable()) {
+        throw new IllegalArgumentException(String.format("Field '%s' is of type array. " +
```

Comment:
similar comments about error messages. Is there no way to consolidate the similar logic?

Reply:
Consolidated this part.